library/core/src/num/ranged.rs

```diff
@@ -0,0 +1,96 @@
+//! Definitions for `Ranged` type.
+
+use crate::ops::{CoerceUnsized, DispatchFromDyn, RangeInclusive};
+
+#[cfg_attr(not(bootstrap), lang = "ranged")]
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+#[repr(transparent)]
+#[unstable(feature = "ranged_int", issue = "none")]
+/// A type that can only represent types in the given range.
+/// Layout optimizations will take.
+pub struct Ranged<T, const RANGE: RangeInclusive<u128>>(T);
```

Comment:
Why are the range boundaries always `u128`, rather than the same type `T`?

Using `u128` prevents setting ranges that include negative numbers, and may be insufficient if a `u256` type is added in the future.

Reply:
This is explained in the main post of this PR